pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -354,11 +355,36 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	// Remove dead container log symlinks.
 	// TODO(random-liu): Remove this after cluster logging supports CRI container log path.
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
+	var containers []*runtimeapi.Container
+	if len(logSymlinks) > 0 {
+		var err error
+		if containers, err = cgc.manager.getKubeletContainers(true); err != nil {
+			klog.V(5).Infof("unable to obtain containers: %v", err)
+			return nil
+		}
+	}
+outer:
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
+			if containerID, err := getContainerIDFromLegacyLogSymlink(logSymlink); err == nil {
```

Comment:
It is my bad, sorry. Feel free to change back to `ContainerStatus`, which is cleaner here.